src/mutex.h

```diff
@@ -2,32 +2,16 @@
 #define BENCHMARK_MUTEX_H_
 
 #include <condition_variable>
-#include <mutex>
 
+#include "absl/synchronization/mutex.h"
 #include "check.h"
 
-// Enable thread safety attributes only with clang.
-// The attributes can be safely erased when compiling with other compilers.
-#if defined(HAVE_THREAD_SAFETY_ATTRIBUTES)
-#define THREAD_ANNOTATION_ATTRIBUTE_(x) __attribute__((x))
-#else
-#define THREAD_ANNOTATION_ATTRIBUTE_(x)  // no-op
-#endif
+#define THREAD_ANNOTATION_ATTRIBUTE_ THREAD_ANNOTATION_ATTRIBUTE__
```

Comment:
Not needed?

Reply:
keeps the changes below simpler. If you prefer, I can remove it and replace THREAD_ANNOTATION_ATTRIBUTE_ with THREAD_ANNOTATION_ATTRIBUTE__

Reply:
if it's not too much trouble, perhaps it'd be great to do that (since you're already overhauling things here) :)
(Optional request, though)